sycl/doc/SYCLInstrumentationUsingXPTI.md

```diff
@@ -0,0 +1,244 @@
+| `wait_end`         | <li>**trace_type**: `xpti::trace_point_type_t::wait_end` that marks the beginning of the wait on an `event`</li> <li> **parent**: `nullptr`</li> <li> **event**: The event ID will reflect the ID of the command group object submission that created this event or the graph event, if the event is an external event. </li> <li> **instance**: Unique ID to allow the correlation of the `wait_begin` event with the `wait_end` event. </li> <li> **user_data**: String indicating `queue.wait` and the address of the event sent in as `const char *` </li>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | None                                                                                                                                                                                                                                                     |
+| `wait_begin`       | <li>**trace_type**: `xpti::trace_point_type_t::wait_begin` that marks the beginning of the wait on an `event`</li> <li> **parent**: `nullptr`</li> <li> **event**: The event ID will reflect the ID of the command group object submission that created this event or a new event based on the combination of the string "queue.wait" and the address of the event.  </li> <li> **instance**: Unique ID to allow the correlation of the `wait_begin` event with the `wait_end` event. </li> <li> **user_data**: String indicating `event.wait` and the address of the event sent in as `const char *` </li> <p></p>Tracing the `queue.wait()` or `queue.wait_and_throw()` will capture the waiting on the action represented by the event object, which could be the execution of a kernel, completion of a memory operation, etc that is embedded in the command group handler. All wait events contain metadata that indicates the SYCL device on which the corresponding operation has been submitted. If the event is from a command group handler, then the source location information is available as well.    | **`sycl_device`**, `sym_function_name`, `sym_source_file_name`, `sym_line_no`                                                                                                                                                                            |
```

Comment:
Are these supposed to be `barrier_begin` and `barrier_end`?

Reply:
Yes, wait_begin/wait_end are used to model barriers. In SYCL, these events are generated from event.wait() and similar calls.

Reply:
@tovinkere I mean wait_begin and wait_end are repeated twice, but barriers, that are also [reported](https://github.com/intel/llvm/blob/7735139bb4d8420fcde692b6eddd2ebf9257efe8/sycl/source/detail/scheduler/commands.cpp#L607) are not mentioned.

Reply:
Ah, thanks - good catch! barrier_begin/barrier_end trace events were not documented. wait_begin/wait_end has two flavors and they have been documented, but I had to make some changes to add clarity.